src/file/aedat4.hpp

```diff
@@ -372,19 +164,276 @@ struct AEDAT4 {
+
+    //   auto attributes = collect_attributes(outinfo);
+    //   if (attributes["name"] != "outInfo") {
+    //     continue;
+    //   }
+
+    //   for (rapidxml::xml_node<> *child = outinfo->first_node(); child;
+    //        child = child->next_sibling()) {
+    //     OutInfo info;
+    //     auto attributes = collect_attributes(child);
+    //     if (!attributes.contains("name")) {
+    //       continue;
+    //     }
+
+    //     info.name = std::stoi(attributes["name"]);
+
+    //     for (rapidxml::xml_node<> *attr = child->first_node(); attr;
+    //          attr = attr->next_sibling()) {
+    //       auto attributes = collect_attributes(attr);
+    //       if (attributes["key"] == "compression") {
+    //         info.compression = attr->value();
+    //       } else if (attributes["key"] == "typeIdentifier") {
+    //         info.type = OutInfo::to_type(attr->value());
+    //       } else if (attributes["name"] == "info") {
+    //         for (rapidxml::xml_node<> *info_node = attr->first_node();
+    //              info_node; info_node = info_node->next_sibling()) {
+    //           auto infos = collect_attributes(info_node);
+
+    //           if (infos["key"] == "sizeX") {
+    //             info.size_x = std::stoi(info_node->value());
+    //           } else if (infos["key"] == "sizeY") {
+    //             info.size_y = std::stoi(info_node->value());
+    //           }
+    //         }
+    //       }
+    //     }
+    //     outinfos.push_back(info);
+    //   }
+    // }
+
+    // TODO: Re-introduce XML parsing
+
+    // we have to treat each packet according the compression method used,
+    // which can be found in ioheader->compression()
+    // assume LZ4 compression for now
+    LZ4F_errorCode_t lz4_error =
+        LZ4F_createDecompressionContext(&ctx, LZ4F_VERSION);
+
+    if (LZ4F_isError(lz4_error)) {
+      const auto message = "Error creating LZ4 decompression context: " +
+                           std::string(LZ4F_getErrorName(lz4_error));
+      throw std::runtime_error(message);
+    }
+
+    // Load data table
+    fseek(fp.get(), data_table_position, SEEK_SET);
+    size_t data_table_size = stat_info.st_size - data_table_position;
+    size_t table_ret = fread(data, 1, data_table_size, fp.get());
+    if (table_ret != data_table_size) {
+      throw std::runtime_error("Failed to read AEDAT4 data table");
+    }
+    auto ret = LZ4F_decompress(ctx, &dst_buffer[0], &dst_size, data,
+                               &data_table_size, nullptr);
+    if (LZ4F_isError(ret)) {
+      std::string message = "Error decompressing AEDAT4 DataTable:" +
+                            std::string(LZ4F_getErrorName(ret));
+      throw std::runtime_error(message);
+    }
+    const auto table = GetSizePrefixedFileDataTable(&dst_buffer[0]);
+
+    // Record offsets of event packets
+    // TODO: add IMU + frames
+    auto packets = table->table();
+    for (size_t i = 0; i < packets->size(); i++) {
```

Comment:
It's just that the post-increment operator creates a temporary (save the current value, increment and return the old value), while and the pre-increment doesn't (it just increments the value). So the post-increment can have a minor performance impact (if you have millions of steps) without any benefit because we are not using the pre-increment value. As I said, very much a nitpick - maybe the compiler can even optimise this out?

Reply:
Wow, I had no idea. I'll fix that and update - thank you for that information!